File: app/src/main/java/com/gustu/logbook/login/presenter/LoginPresenter.java
package com.gustu.logbook.login.presenter;

import com.gustu.logbook.login.interfaces.LoginView;
import com.gustu.logbook.login.model.LoginResponse;
import com.gustu.logbook.network.loginnetwork.LoginBaseUrl;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginPresenter {
    LoginView loginView;
    LoginBaseUrl loginBaseUrl;

    public LoginPresenter(LoginView loginView) {
        this.loginView = loginView;
        if (this.loginBaseUrl==null){
            this.loginBaseUrl= new LoginBaseUrl();
        }
    }
    public void gotoLogin(String username , String password){
        loginBaseUrl.getLoginApi().login(username, password).enqueue(new Callback<LoginResponse>() {
            @Override
            public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
                if (response.isSuccessful()){
                    LoginResponse loginResponse = response.body();
                    loginView._onLogin(loginResponse);
                }
            }

            @Override
            public void onFailure(Call<LoginResponse> call, Throwable t) {
                    loginView._onFailed();
            }
        });
    }
}

File: app/src/main/java/com/gustu/logbook/login/presenter/LoginPresenter.kt
package com.gustu.logbook.login.presenter

import android.util.Log

import com.gustu.logbook.login.interfaces.LoginView
import com.gustu.logbook.login.model.LoginResponse
import com.gustu.logbook.network.loginnetwork.LoginBaseUrl

import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class LoginPresenter(internal var loginView: LoginView) {
    internal var loginBaseUrl: LoginBaseUrl? = null

    init {
        if (this.loginBaseUrl == null) {
            this.loginBaseUrl = LoginBaseUrl()
        }
    }

    fun gotoLogin(username: String, password: String) {
        Log.d("LoginPresenter", "gotoLogin: $username$password")
        loginBaseUrl!!.loginApi.login(username, password).enqueue(object : Callback<LoginResponse> {
            override fun onResponse(call: Call<LoginResponse>, response: Response<LoginResponse>) {
                if (response.isSuccessful) {
                    val loginResponse = response.body()
                    loginView._onLogin(loginResponse!!)
                }
            }

            override fun onFailure(call: Call<LoginResponse>, t: Throwable) {
                loginView._onFailed()
            }
        })
    }
}
